### ISSUE TITLE
Add forward delete behavior to delete key

### ISSUE THREAD

**Issue by Hx2-code** [florisboard/florisboard#450]
In Samsung keyboard I found a feature to delete text from left to right by pressing on the capslock icon and then pressing the delete icon while holding the capslock.
A feature like this on FlorisBoard would be great.

**Comment by patrickgold** [comment]
Agreed, this sounds like a cool feature. Will add it on my todo list!

**Comment by Perdellian** [comment]
[this comment](https://github.com/florisboard/florisboard/issues/464#issuecomment-808936781) on #464 suggests using the shift key to swipe-delete forward as well

Would it be feasible to (optionally?) extend this to reversing the direction of the delete key in general, such that it would also apply when _Gestures & Glide typing_ > _Delete key long press_ is set to "Delete word before cursor"?

**Comment by patrickgold** [comment]
Implementation idea:

delete key | shift? | behavior
---|---|---
tap | no | deletes 1 char/word before cursor
tap | yes | deletes 1 char/word after cursor
tap&hold | no | deletes N chars/words fast before cursor
tap&hold | yes | deletes N chars/words fast after cursor
swipe | no | selects N chars/words and deletes on release before cursor
swipe | yes | selects N chars/words and deletes on release after cursor

This makes the key very versatile, while not taking any more space on the keyboard. Behavior is also compatible with RTL languages.

Open issues: how should the key behave if shift? state changes during a swipe?

**Comment by mysteryoneal** [comment]
#2042 was not a duplicate of this. 2042 was requesting a separate key that has forward delete functionality, not a key combination like this thread is talking about. You need to reopen the 2042.

**Comment by patrickgold** [comment]
PR #3010 adds a combined shift+delete key behavior which acts as a forward delete, which should fix this issue.
